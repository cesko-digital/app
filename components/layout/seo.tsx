/* eslint-disable @next/next/no-css-tags */
import Head from "next/head";
import strings from "content/strings.json";

export interface SeoProps {
  description?: string;
  title?: string;
  coverUrl?: string;
}

export const Seo: React.FC<SeoProps> = ({
  description,
  title,
  coverUrl = "https://data.cesko.digital/web/metadata-cover.png",
}: SeoProps) => {
  return (
    <Head>
      <title>
        {title ? `${title} | ÄŒesko.Digital` : strings.metadata.title}
      </title>
      <meta
        name="description"
        content={description || strings.metadata.description}
      />
      <meta name="og:image" content={coverUrl} />
      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:image" content={coverUrl} />
      <link rel="stylesheet" href="/fonts.css" />
    </Head>
  );
};

export default Seo;
