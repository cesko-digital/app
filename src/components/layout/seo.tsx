import React, { useContext } from 'react'
import Helmet from 'react-helmet'
import { I18nextContext, useTranslation } from 'gatsby-plugin-react-i18next'
import { insertIf } from 'utils/insert-if'

export interface SeoProps {
  description?: string
  title?: string
  coverUrl?: string
}

export const Seo: React.FC<SeoProps> = ({
  description,
  title,
  coverUrl = 'https://data.cesko.digital/web/metadata-cover.png',
}: SeoProps) => {
  const { language } = useContext(I18nextContext)
  const { t } = useTranslation()
  const metaDescription = description || t('metadata.description')

  const metadataTitle = t('metadata.title')
  return (
    <Helmet
      htmlAttributes={{
        lang: language,
      }}
      title={title}
      defaultTitle={metadataTitle}
      titleTemplate={'%s | ÄŒesko.Digital'}
      meta={[
        {
          name: 'description',
          content: metaDescription,
        },
        {
          property: 'og:title',
          content: title,
        },
        {
          property: 'og:description',
          content: metaDescription,
        },
        {
          property: 'og:type',
          content: 'website',
        },
        {
          property: 'og:image',
          content: coverUrl,
        },
        {
          name: 'twitter:card',
          content: 'summary_large_image',
        },
        {
          name: 'twitter:creator',
          content: metadataTitle,
        },
        {
          name: 'twitter:title',
          content: title,
        },
        {
          name: 'twitter:description',
          content: metaDescription,
        },
        {
          name: 'twitter:image',
          content: coverUrl,
        },
        ...insertIf({
          condition: process.env.GATSBY_NO_INDEX === 'true',
          elements: [{ name: 'robots', content: 'noindex' }],
        }),
      ].concat()}
    />
  )
}

export default Seo
