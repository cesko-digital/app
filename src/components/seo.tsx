/**
 * SEO component that queries for data with
 *  Gatsby's useStaticQuery React hook
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from 'react'
import Helmet from 'react-helmet'
import { useStaticQuery, graphql } from 'gatsby'

interface PureSEOProps {
  data: any
  description?: string
  lang?: string
  meta?: []
  title?: string
}

interface SEOProps {
  description?: string
  lang?: string
  meta?: []
  title: string
}

export const PureSEO = ({
  data,
  description,
  lang,
  meta,
  title,
}: PureSEOProps) => {
  const metaDescription = description || data.site.siteMetadata.description

  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={title}
      titleTemplate={`%s | ${data.site.siteMetadata.title}`}
      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: data.site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
      ].concat()}
    />
  )
}

const SEO = ({ description, lang, meta, title }: SEOProps) => {
  const data = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
            description
            author
          }
        }
      }
    `
  )

  return <PureSEO data={data} description={description} />
}

export default SEO
