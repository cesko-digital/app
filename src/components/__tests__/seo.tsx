import React from 'react'
import renderer from 'react-test-renderer'
import { GraphQLData, PureSEO as SEO } from '../seo'

describe('SEO', () => {
  it('renders correctly', () => {
    const data: GraphQLData = {
      site: {
        siteMetadata: {
          title: 'cesko.digital',
          description: 'lorem ipsum',
          author: 'Jose',
        },
      },
    }
    const tree = renderer
      .create(<SEO data={data} title={data.site.siteMetadata.title} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
