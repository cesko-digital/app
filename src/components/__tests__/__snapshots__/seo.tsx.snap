// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders correctly 1`] = `null`;
